packages/schema-derive/Cargo.toml

```diff
@@ -8,9 +8,11 @@ repository = "https://github.com/CosmWasm/cosmwasm/tree/main/packages/schema"
 license = "Apache-2.0"
 
 [dependencies]
+heck = "0.5.0"
+manyhow = "0.11.1"
```

Comment:
I like that you're doing something nicer than the panics.

The `manyhow` dependency makes me apprehensive TBH. I don't think we currently have anyone passionate about maintaining this macro crate. At a glance, the `manyhow` thing is fairly young, doesn't seem to be widely used in the ecosystem, and (probably) broke its API twice in the last 4 months.

To keep things more stable, do you think we could use the APIs provided by `syn` for error handling instead? Or is that not sufficient?

Reply:
Actually, I think we can replace these pretty easily with some home-grown declarative macros. Shouldn't be too hard

Reply:
There we go. 40 lines of declarative macros, and almost no code changes (except for one weird borrow we need now)